hail/src/main/scala/is/hail/io/tabix/TabixReader.scala

```diff
@@ -370,33 +375,27 @@ final class TabixLineIterator(
         bufferCursor += 1
       }
 
-      if (bufferCursor >= bufferLen) {
+      if (bufferCursor == bufferLen) {
       // no newline before end of buffer
         if (bufferEOF) {
           isEof = true
           val str = decodeString(start, bufferCursor)
           bufferCursor += 1
           return str
+        } else if (bufferLen == buffer.length) {
```

Comment:
Can you explain why we need this restructuring? The previous logic (importantly, the bufferCursor == 0 check) is designed to prevent unbounded growth in the buffer, which I'm not convinced this approach avoids.

Reply:
Under what circumstances would `bufferCursor` be 0 in the previous code? As far as I can tell, that branch would be taken never.

Reply:
I'm pretty sure the previous branch could never be taken. Under what circumstances could `bufferCursor == 0` and `bufferCursor == bufferLen`?

Reply:
hah, oops. That should have been `if (start == 0)`.

Reply:
you make a good point :D

Reply:
that assertion is probably also invalid (assert bufferLen == buffer.length)

Reply:
probably, I still think my version is cleaner with what the cases actually mean, with less conditional nesting.

I'm also pretty sure that this code as I've written it will essentially grow the buffer to at most 2x the size of the longest line in the VCF slice we're looking at, which is some of the best we're going to do. Another thing we could do is grow it 64k at a time, because reading is probably slower than reallocating or memmoving. :shrug: